src/pykeen/models/base.py

```diff
@@ -301,18 +314,145 @@ def load_state(self, path: Union[str, os.PathLike]) -> None:
         """
         self.load_state_dict(torch.load(path, map_location=self.device))
 
-    """Prediction methods"""
+    """Extended scoring methods"""
+
+    def _prepare_batch(
+        self,
+        batch: Optional[torch.LongTensor],
+        index_relation: int,
+        invert_relation: bool,
+    ) -> torch.LongTensor:
+        if invert_relation and not self.use_inverse_relations:
+            raise ValueError("Can only invert relations if use_inverse_relations is set to True")
+
+        # TODO: with the current default inversion, we have to materialize the relation IDs
```

Comment:
TODO: with the current default inversion, we have to materialize the relation IDs